koku/masu/processor/report_parquet_processor_base.py

```diff
@@ -124,18 +124,29 @@ def _generate_create_table_sql(self):
             sql += f"{norm_col} {col_type}"
             if idx < (len(parquet_columns) - 1):
                 sql += ","
-        sql += ",source varchar, year varchar, month varchar"
+        if partition_map:
+            # Add the specified partition columns
+            sql += ", "
+            sql += ",".join([f"{item[0]} {item[1]} " for item in list(partition_map.items())])
+            partition_column_str = ",".join([f"'{key}' " for key in partition_map.keys()])
+            sql += (
+                f") WITH(external_location = 's3a://{s3_path}', format = 'PARQUET',"
+                f" partitioned_by=ARRAY[{partition_column_str}])"
+            )
+        else:
+            # Use the default partition columns
+            sql += ",source varchar, year varchar, month varchar"
 
-        sql += (
-            f") WITH(external_location = 's3a://{s3_path}', format = 'PARQUET',"
-            " partitioned_by=ARRAY['source', 'year', 'month'])"
-        )
+            sql += (
+                f") WITH(external_location = 's3a://{s3_path}', format = 'PARQUET',"
+                " partitioned_by=ARRAY['source', 'year', 'month'])"
+            )
         LOG.info(f"Create Parquet Table SQL: {sql}")
         return sql
 
-    def create_table(self):
+    def create_table(self, partition_map=None):
         """Create Trino SQL table."""
-        sql = self._generate_create_table_sql()
+        sql = self._generate_create_table_sql(partition_map=partition_map)
         self._execute_sql(sql, self._schema_name)
```

Comment:
Just curious if we are worried about SQL injection attacks here. It looks like most of the values are not coming from direct user input, but I'm not 100% sure.

Reply:
I don't believe we have to be concerned with SQL injection here, the partition map is statically defined by us and I believe the rest of the _generate_create_table_sql is also values we have defined elsewhere that we base on an expected list such as column names. I don't believe we use any user defined input here.